shared/js/https.js

```diff
@@ -1,228 +1,211 @@
+            return reqUrl
         }
 
-        if (this.isSync) {
-            return checkDb.call(this)
-        } else {
-            return new Promise((resolve) => checkDb.call(this, resolve))
+        // Only deal with http calls
+        const protocol = utils.getProtocol(reqUrl).toLowerCase()
+        if (!protocol.indexOf('http:') === 0) {
+            return reqUrl
         }
-    }
 
-    /**
-     * For debugging/development/test purposes only
-     * Logs all records in indexed db to console
-     */
-    logAllRecords () {
-        this.db.logAllRecords(this.dbObjectStore)
-    }
+        // Obey global settings (options page)
+        if (!settings.getSetting('httpsEverywhereEnabled')) {
+            return reqUrl
+        }
 
-    /**
-     * For debugging/development/test purposes only
-     * This only tests ability to retrieve record from indexed db
-     * Usage: .getHostRecord('goodreads.com').then((r) => console.log(r))
-     */
-    getHostRecord (host) {
-        return new Promise ((resolve, reject) => {
-            if (!this.isReady) {
-                console.warn('HTTPS: this.db is not ready')
-                return reject()
-            }
+        // Skip upgrading sites that have been whitelisted by user
+        // via on/off toggle in popup
+        if (tab.site.whitelisted) {
+            console.log(`HTTPS: ${tab.site.domain} was whitelisted by user. skip upgrade check.`)
+            return reqUrl
+        }
 
-            this.db.get(this.dbObjectStore, host).then(
-                (record) => {
-                    if (record) return resolve(record)
-                    return resolve()
-                },
-                () => {
-                    console.warn('HTTPS: getHostRecord() encountered a db error.')
-                    return reject()
-                }
-            )
+        // Determine host
+        const host = utils.extractHostFromURL(reqUrl)
+        const hosts = [host]
 
-        })
-    }
+        // Check if host has an entry as a wildcarded subdomain
+        const subdomain = utils.extractTopSubdomainFromHost(host)
+        if (subdomain && subdomain !== 'www') {
+            const wildcard = host.replace(subdomain, '*')
+            hosts.push(wildcard)
+        }
 
-    /* For debugging/development/test purposes only
-     * Tests the .getHostRecord() method against an array of test hosts
-     * defined below
-     */
-    testGetHostRecord (cb) {
-        // These hosts should always have records that were xhr'd
-        // into the client-side db from server
-        const testHosts = [
-            '1337x.to',
-            'submit.pandora.com',
-            '*.api.roblox.com',
-            'thump.vice.com',
-            'yts.ag'
-        ]
+        // Check for upgrades
+        for (let i=0; i<hosts.length; i++) {
+            if (httpsUpgradeList.indexOf(hosts[i]) > -1) {
+                return reqUrl.replace(/^(http|https):\/\//i, 'https://')
```

Comment:
So we only operate on the front of the URL here. I wish we had a better way if we had to make further modifications like subdomains and path fragments. I'll look but assume it was the same before.